tool/mapeditor.re

```diff
@@ -50,60 +65,102 @@ static MAP_EDITOR_COMPONENT = Component {
     on_menu_open: fn() {},
 };
 
-static mut MAP_EDITOR_LABEL = Text { text: if CURRENT_COMPONENTS.contains(MAP_EDITOR_COMPONENT) { "Stop Map Editor" } else { "Edit Map" } };
+static mut MAP_EDITOR_EDIT_LABEL = Text { text: if MAP_EDITOR_STATE.mode != MapEditorMode::None { "Stop Map Editor" } else { "Edit Map" } };
+static mut MAP_EDITOR_PLAY_LABEL = Text { text: if CURRENT_COMPONENTS.contains(MAP_EDITOR_COMPONENT) { "Stop Map Editor" } else { "Play Map" } };
```

Comment:
Those don't need to be statics anymore. They only need to be statics when their value can change while you stay within the menu. For these buttons, you leave the menu when pressed.